test/e2e_env/multizone/meshtimeout/meshtimeout.go

```diff
@@ -259,4 +269,50 @@ spec:
 			g.Expect(response.ResponseCode).To(Equal(504))
 		}, "30s", "1s").Should(Succeed())
 	})
+
+	It("should apply MeshTimeout policy on MeshService from other zone", func() {
+		// given
+		// create a tunnel to test-client admin
+		Expect(multizone.KubeZone1.PortForwardService("test-client", k8sZoneNamespace, 9901)).To(Succeed())
+		portFwd := multizone.KubeZone1.GetPortForward("test-client")
+		tnl := tunnel.NewK8sEnvoyAdminTunnel(multizone.Global.GetTesting(), portFwd.ApiServerEndpoint)
+
+		Eventually(func(g Gomega) {
+			_, err := framework_client.CollectEchoResponse(
+				multizone.KubeZone1, "test-client", "http://test-server.multizone-meshtimeout-ns.svc.kuma-2.mesh.local:80",
+				framework_client.FromKubernetesPod(k8sZoneNamespace, "test-client"),
+			)
+			g.Expect(err).ToNot(HaveOccurred())
+		}, "30s", "1s").MustPassRepeatedly(5).Should(Succeed())
+		// should have active connection
+		Consistently(func(g Gomega) {
+			g.Expect(activeCxStat(tnl)).To(stats.BeGreaterThanZero())
+		}, "10s", "1s").Should(Succeed())
```

Comment:
10s is pretty long wait, it'd be nice if we could avoid it. Maybe you can check the `upstream_cx_idle_timeout` stat to be 0 before policies is applied and greater than 0 after.

Reply:
Changed the timeout to 5 seconds. Checking only 0 doesn't make sense because if the connection closes right after the check, the result might not be reliable. I reduced the time to 5 seconds and updated the stat accordingly.

Reply:
`upstream_cx_idle_timeout` seems not stable and flaky. `cx_active` works better